pkg/liqonet/ipam.go

```diff
@@ -188,34 +208,45 @@ func (liqoIPAM *IPAM) GetSubnetPerCluster(network, clusterID string) (string, er
 		}
 		if err == nil {
 			klog.Infof("Network %s successfully assigned to cluster %s", network, clusterID)
-			liqoIPAM.SubnetPerCluster[clusterID] = network
+			ipamConfig.Spec.ClusterSubnet[clusterID] = network
+			if err := liqoIPAM.ipamStorage.updateClusterSubnet(ipamConfig.Spec.ClusterSubnet); err != nil {
+				return "", err
+			}
 			return network, nil
 		}
 		/* Network is not available, need a mapping */
 		klog.Infof("Cannot acquire network %s from pool %s", network, pool)
 	}
 	/* Network is already reserved, need a mapping */
 	klog.Infof("Looking for a mapping for network %s...", network)
-	mappedNetwork, ok = liqoIPAM.mapNetwork(network)
-	if !ok {
+	mappedNetwork, err = liqoIPAM.mapNetwork(network)
+	if err != nil {
 		return "", fmt.Errorf("Cannot assign any network to cluster %s", clusterID)
 	}
 	klog.Infof("Network %s successfully mapped to network %s", mappedNetwork, network)
 	klog.Infof("Network %s successfully assigned to cluster %s", mappedNetwork, clusterID)
-	liqoIPAM.SubnetPerCluster[clusterID] = mappedNetwork
+	ipamConfig.Spec.ClusterSubnet[clusterID] = mappedNetwork
+	if err := liqoIPAM.ipamStorage.updateClusterSubnet(ipamConfig.Spec.ClusterSubnet); err != nil {
+		return "", err
+	}
 	return mappedNetwork, nil
 }
 
 // mapNetwork allocates a suitable (same mask length) network used to map the network received as first parameter which probably cannot be used due to some collision with existing networks
-func (liqoIPAM *IPAM) mapNetwork(network string) (string, bool) {
-	for _, pool := range liqoIPAM.pools {
-		klog.Infof("Trying to acquire a child prefix from prefix %s (mask length=%d)", pool, getMask(network))
+func (liqoIPAM *IPAM) mapNetwork(network string) (string, error) {
+	// Get resource
+	ipamConfig, err := liqoIPAM.ipamStorage.getConfig()
+	if err != nil {
+		return "", fmt.Errorf("cannot get Ipam config: %w", err)
+	}
+	for _, pool := range ipamConfig.Spec.Pools {
+		klog.Infof("Trying to acquire a child prefix from prefix %s (mask lenght=%d)", pool, getMask(network))
 		if mappedNetwork, err := liqoIPAM.ipam.AcquireChildPrefix(pool, getMask(network)); err == nil {
 			klog.Infof("Network %s has been mapped to network %s", network, mappedNetwork)
-			return mappedNetwork.String(), true
+			return mappedNetwork.String(), nil
 		}
 	}
-	return "", false
+	return "", fmt.Errorf("there are no more networks available")
```

Comment:
For the sake of coherence, the message should start with a capital letter.

Reply:
Golang returns a warning if I try to start an Error with a capital letter. For this reasons, all errors I've ever wrote start with a lowercase letter. Should I change each of them?